### PR TITLE
Run update-packages --force-upgrade

### LINKED ISSUE #94060: [tool] flutter doctor throws ` FormatException: Unexpected extension byte` when using `IntelliJ IDEA  2021.3`  and `Dart` plugin
## Steps to Reproduce

1. Install new IntelliJ IDEA 2021.3 RC and Dart plugin
2. Run `flutter doctor`

**Expected results:** 

No problem with doctor

**Actual results:** 

```
[☠] IntelliJ IDEA Community Edition (the doctor check crashed)
    ✗ Due to an error, the doctor check did not complete. If the error message below is not helpful, please let us know about this issue at https://github.com/flutter/flutter/issues.
    ✗ FormatException: Unexpected extension byte (at offset 5)
```

The crash come from https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/intellij/intellij.dart#L130 with `Dart.jar` as file

<details>
  <summary>Logs</summary>

```
flutter doctor 
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel master, 2.6.0-12.0.pre.790, on Fedora Linux 35 (KDE Plasma) 5.14.18-300.fc35.x86_64, locale fr_FR.UTF-8)
[!] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    ✗ cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/linux#android-setup for more details.
[✓] Chrome - develop for the web
[✓] Linux toolchain - develop for Linux desktop
[✓] Android Studio (version 2020.3)
[✓] IntelliJ IDEA Community Edition (version 2021.2)
[☠] IntelliJ IDEA Community Edition (the doctor check crashed)
    ✗ Due to an error, the doctor check did not complete. If the error message below is not helpful, please let us know about this issue at https://github.com/flutter/flutter/issues.
    ✗ FormatException: Unexpected extension byte (at offset 5)
[✓] VS Code (version 1.62.3)
[✓] Connected device (2 available)

! Doctor found issues in 2 categories.

```

### PR DESCRIPTION
**Pull request by jmagman**
Update packages.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: dev/tools/gen_defaults/pubspec.yaml [hunk 1]
```diff
@@ -7,34 +7,31 @@ environment:
...
 dependencies:
 
-  async: 2.8.2 # THIS LINE IS AUTOGENERATED - TO UPDATE USE "flutter update-packages --force-upgrade"
-  charcode: 1.3.1 # THIS LINE IS AUTOGENERATED - TO UPDATE USE "flutter update-packages --force-upgrade"
-  collection: 1.15.0 # THIS LINE IS AUTOGENERATED - TO UPDATE USE "flutter update-packages --force-upgrade"
-  http_parser: 4.0.0 # THIS LINE IS AUTOGENERATED - TO UPDATE USE "flutter update-packages --force-upgrade"
-  source_span: 1.8.1 # THIS LINE IS AUTOGENERATED - TO UPDATE USE "flutter update-packages --force-upgrade"
-  string_scanner: 1.1.0 # THIS LINE IS AUTOGENERATED - TO UPDATE USE "flutter update-packages --force-upgrade"
-  term_glyph: 1.2.0 # THIS LINE IS AUTOGENERATED - TO UPDATE USE "flutter update-packages --force-upgrade"
-  typed_data: 1.3.0 # THIS LINE IS AUTOGENERATED - TO UPDATE USE "flutter update-packages --force-upgrade"
```
Comment (jmagman) [inline dev/tools/gen_defaults/pubspec.yaml:17]: @darrenaustin looking at https://github.com/flutter/flutter/pull/95370 this change looks correct, but I'm not sure why these packages were included in `dependencies` in the first place.  Did you create this pubspec manually, or did you run `flutter update-packages` and `AUTOGENERATED` was copied and pasted from somewhere?

Comment (darrenaustin) [reply dev/tools/gen_defaults/pubspec.yaml:17]: This looks fine. The only dependencies this tool has is the two dev_dependencies `path` and `test`.

I was having a problem getting this pubspec into the tree. I wound up copying entries from gen_keycodes's pubspec to get it to pass the analyzer. Sorry this is causing confusion. What is the correct way to add a new pubspec to the tree? Thx.

**Comment by TahaTesser** [comment]
`archive` update is required to fix https://github.com/flutter/flutter/issues/94060
I just confirmed it locally https://github.com/flutter/flutter/issues/94060#issuecomment-1008667631

File: dev/tools/gen_defaults/pubspec.yaml (hunk 1 above)
Comment (jmagman) [reply dev/tools/gen_defaults/pubspec.yaml:17]: I guess what you did, and then an additional step of running `update-packages --force-upgrade` to match the other pubspec versions and clean up.

**Review by Piinks — APPROVED** [review]
So nice, LGTM. :)

**Review by christopherfujino — APPROVED** [review]
LGTM, Woohoo, excited about fixing the Dart.jar thing!

**Pull request merged**